src/app/pages/datasets/components/dataset-form/name-length-validation.ts

```diff
@@ -0,0 +1,34 @@
+import { UntypedFormControl, ValidatorFn } from '@angular/forms';
+import { DefaultValidationError } from 'app/enums/default-validation-error.enum';
+
+export function datasetNameTooLong(parentPath: string): ValidatorFn {
+  const maxLengthAllowed = 200;
+  let thisControl: UntypedFormControl;
+
+  return function datasetNameTooLongValidate(control: UntypedFormControl) {
+    if (!control.parent) {
```

Comment:
can we merge 3 if blocks in one?

**if (!parentPath || !control.value || !control.parent) {
  return null
}**